### PR TITLE
javascript helpertext

### PR DESCRIPTION
**Pull request by the-daniel-rothig**

### CODE REVIEW

**Comment by the-daniel-rothig** [comment]
I disagree with the codacy failure, and would like to ignore the checks. If you disagree please merge :)

**Comment by the-daniel-rothig** [comment]
this will conflict with pull request 28, so please merge that one before this one - I'll fix it in-branch as soon as 28 is merged

**Comment by DougC** [comment]
So `FieldRule` no longer exists - replaced with `FieldValidation`, and I made some changes to the renderers that hopefully won't have created too-large a conflict.

File: src/main/public/javascripts/application-rules.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+(function() {
+    if(!document.getElementsByClassName || !document.body.addEventListener) {return;}
```
Comment (jon-shipley) [inline src/main/public/javascripts/application-rules.js]: Why not `use strict` here?  E.g. `"use strict";`

**Comment by the-daniel-rothig** [comment]
HI @DougC and @jon-shipley, this now works with The Big Refactor. I modified FieldHint to carry some metadata from FieldValidation that the JS Hint text can use to initialise itself

**Comment by jon-shipley** [comment]
This is quite nice :)

However, there is a bug in this PR that prevents it working in IE10 and lower.  I see this has already been merged, so would you mind creating a new PR with the fix?

IE8 code is not working (and won't work even with the fix) - we have agreed to check if IE8 is actually a user requirement before we either a) implement manual work-around or b) use a library to work-around it for us.

**Comment by tomfarrarbis** [comment]
GDS make IE8 sound like it's not an optional, currently. Hopefully this will change because OpenShift also doesn't support IE8 on XP (because of a lack of SNI support etc).

4 or 5 months ago roughly ~4% of service users were utilising IE8, FWIW. This will need checking again, but it is out there.

**Comment by the-daniel-rothig** [comment]
@jon-shipley fixed in pr 30 - thanks for spotting this. Re: IE8, I think we are *not* talking about dropping IE8, but applying graceful degradation to IE8. If you look at how GOV.UK layouting works* in IE8 and lower, the reactiveness of hint text is not the most degraded element by a long shot ;). 

*or rather, doesnt

**Pull request merged**